server/internal/infprocessor/infprocessor.go

```diff
@@ -0,0 +1,136 @@
+package infprocessor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/common/pkg/models"
+)
+
+const (
+	taskQueueSize  = 100
+	completionPath = "/v1/chat/completions"
+)
+
+// Task is an infeference task.
+type Task struct {
+	Req    *v1.CreateChatCompletionRequest
+	Header http.Header
+	RespCh chan *http.Response
+	ErrCh  chan error
+}
+
+// WaitForCompletion waits for the completion of the task.
+func (t *Task) WaitForCompletion() (*http.Response, error) {
+	select {
+	case resp := <-t.RespCh:
+		return resp, nil
+	case err := <-t.ErrCh:
+		return nil, err
+	}
+}
+
+// NewTaskQueue creates a new task queue.
+func NewTaskQueue() *TaskQueue {
+	return &TaskQueue{
+		tasks: make(chan *Task, taskQueueSize),
+	}
+}
+
+// TaskQueue is a queue for inference tasks.
+type TaskQueue struct {
+	tasks chan *Task
+}
+
+// Enqueue inserts a task into the queue.
+func (q *TaskQueue) Enqueue(t *Task) {
+	q.tasks <- t
+}
+
+// Dequeue removes a task from the queue.
+func (q *TaskQueue) Dequeue(ctx context.Context) (*Task, error) {
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case t := <-q.tasks:
+		return t, nil
+	}
+}
+
+type engineGetter interface {
+	GetEngineForModel(ctx context.Context, modelID string) (string, error)
+}
+
+// NewP creates a new processor.
+func NewP(queue *TaskQueue, engineGetter engineGetter) *P {
+	return &P{
+		queue:        queue,
+		engineGetter: engineGetter,
+	}
+}
+
+// P processes inference tasks.
+type P struct {
+	queue *TaskQueue
+
+	engineGetter engineGetter
+}
+
+// Run runs the processor.
+func (p *P) Run(ctx context.Context) error {
+	for {
+		t, err := p.queue.Dequeue(ctx)
+		if err != nil {
+			return err
+		}
+		p.runTask(ctx, t)
```

Comment:
bypass `runTask` if t is nil?

Reply:
also, it seems a busy-loop, not sure if it should sleep somewhere?

Reply:
Ah `t` should be non-nil when `err` is `nil`. Also Dequeue blocks when there is no task as the channel read is blocking by default.